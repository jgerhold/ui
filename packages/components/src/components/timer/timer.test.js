import { mount } from '@vue/test-utils'

import Timer from './Timer'

describe('Timer', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(Timer)

      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(Timer)

      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'time',
        value: 360
      },
      {
        prop: 'remaining',
        value: true
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(Timer, {
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
