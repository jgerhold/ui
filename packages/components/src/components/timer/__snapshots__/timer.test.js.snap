// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer props should render with prop color 1`] = `
<span
  class="timer"
  style="color: rgb(255, 255, 255);"
>
  00:00
</span>
`;

exports[`Timer props should render with prop remaining 1`] = `
<span
  class="timer"
>
  -00:00
</span>
`;

exports[`Timer props should render with prop time 1`] = `
<span
  class="timer"
>
  00:00
</span>
`;

exports[`Timer props should render without props 1`] = `
<span
  class="timer"
>
  00:00
</span>
`;
